src/collections.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Handy generic functions for operating on JS collections.
+ *
+ * Any function here is the sort of thing that perhaps should be in the
+ * language's standard library.
+ *
+ * See also
+ *  * `jsBackport.js`, for things that actually are in newer versions of JS,
+ *    or late-stage proposals for future versions
+ *  * `generics.js`, for things that operate on types
+ *
+ * @flow strict-local
+ */
+
+/**
+ * Like `Array#map`, but returns null if `f` ever returns a nullish value.
+ *
+ * If `f` returns nullish at any element, the iteration stops and `f` is not
+ * called on any later elements.
+ *
+ * This replaces either (a) writing an explicit loop, in order to be able to
+ * abort on a nullish value, or (b) using `Array#map` and then running
```

Comment:
> abort on a nullish value

~~nit: abort on a null value~~

Hmm! Right, it does abort on a nullish value, not just `null`. The `== null` (`==`, not `===`) makes it abort on a nullish value. I wasn't expecting the double-equals, but I suppose it makes sense to do one operation instead of two (i.e., `r === null || r === undefined`).

Reply:
Yep!

For the API choice, I think "nullish" is most helpful because it means it can work with things like `Map#get` that return undefined as well as with things that, like this function itself, return null.

For the code-style choice: I've gone back and forth, but my current leaning is that it makes sense to use `==` for basically this one pattern, of `== null`.
* It does exactly mean `r === null || r === undefined` -- which makes it just about the only usage of `==` that can be described as cleanly as that.
* It's rather shorter; and it avoids duplicating the expression, which doesn't matter here but can meaningfully reduce mess when it's more complex.
* It's a pretty tightly constrained and bright-line pattern -- so I think it's still quite doable to see `==` in any context other than `== null` and have that stand out as a code smell.

Reply:
(Similarly `!= null`, naturally.)